elements/lisk-cryptography/src/bls_lib/lib.ts

```diff
@@ -43,28 +44,34 @@ export const blsKeyValidate = (pk: Buffer): boolean => {
 // https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-bls-signature-04#section-2.3
 export const blsKeyGen = (ikm: Buffer): Buffer => Buffer.from(SecretKey.fromKeygen(ikm).toBytes());
 
-// check that the secret key generated is non-zero modulo the group order.
-export const isSecretKeyNonZeroModOrder = (secretKey: SecretKey): boolean => {
-	// https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-pairing-friendly-curves-10#section-4.2.1
-	const groupOrder = '0x73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001';
-
-	const skBigInt = BigInt(`0x${Buffer.from(secretKey.toBytes()).toString('hex')}`);
+/**
+ * The function checks if a given buffer `sk` is not a multiple of the group order and 32 bytes long.
+ *
+ * @param {Buffer} sk - The parameter `sk` is a Buffer that represents a secret key.
+ * @returns a boolean value.
+ */
+export const isNotMultipleOfGroupOrder = (sk: Buffer): boolean => {
+	/// https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-pairing-friendly-curves-10#section-4.2.1
+	// r: 0x73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001;
+	const groupOrder = Buffer.from(
+		'73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001',
+		'hex',
+	);
 
-	// check if secret key is non-zero modulo the order of the elliptic curve.
-	if (skBigInt % BigInt(groupOrder) === BigInt(0)) {
-		return false;
-	}
+	// 2 * r (group order)
+	const groupOrderDouble = Buffer.from(
+		'e7db4ea6533afa906673b0101343b00aa77b4805fffcb7fdfffffffe00000002',
+		'hex',
+	);
 
-	return true;
+	return !(timingSafeEqual(sk, groupOrder) || timingSafeEqual(sk, groupOrderDouble));
 };
 
 // https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-bls-signature-04#section-2.4
 export const blsSkToPk = (sk: Buffer): Buffer => {
-	const secretKey = SecretKey.fromBytes(sk);
+	isNotMultipleOfGroupOrder(sk);
```

Comment:
`isNotMultipleOfGroupOrder` returns True / False, so I think you should write `if(!isNotMultipleOfGroupOrder(sk) { raise Exception() }`.